object-store/src/lib.rs

```diff
@@ -514,29 +515,82 @@ impl PyObjectStore {
         Ok(())
     }
 
+    /// Save the provided bytes to the specified location.
+    #[pyo3(text_signature = "($self, location, bytes)")]
+    fn put_async<'a>(
+        &'a self,
+        py: Python<'a>,
+        location: PyPath,
+        bytes: Vec<u8>,
+    ) -> PyResult<&PyAny> {
+        let inner = self.inner.clone();
+        pyo3_asyncio::tokio::future_into_py(py, async move {
+            inner
+                .put(&location.into(), bytes.into())
+                .await
+                .map_err(ObjectStoreError::from)?;
+            Ok(())
+        })
+    }
+
     /// Return the bytes that are stored at the specified location.
     #[pyo3(text_signature = "($self, location)")]
-    fn get(&self, location: PyPath) -> PyResult<Py<PyBytes>> {
+    fn get(&self, location: PyPath) -> PyResult<Cow<[u8]>> {
         let obj = self
             .rt
             .block_on(get_bytes(self.inner.as_ref(), &location.into()))
             .map_err(ObjectStoreError::from)?;
-        Python::with_gil(|py| Ok(PyBytes::new(py, &obj).into_py(py)))
+        Ok(Cow::Owned(obj.to_vec()))
+    }
+
+    /// Return the bytes that are stored at the specified location.
+    #[pyo3(text_signature = "($self, location)")]
+    fn get_async<'a>(&'a self, py: Python<'a>, location: PyPath) -> PyResult<&PyAny> {
+        let inner = self.inner.clone();
+        pyo3_asyncio::tokio::future_into_py(py, async move {
+            let obj = get_bytes(inner.as_ref(), &location.into())
+                .await
+                .map_err(ObjectStoreError::from)?;
+            Ok(Cow::<[u8]>::Owned(obj.to_vec()))
+        })
     }
 
     /// Return the bytes that are stored at the specified location in the given byte range
     #[pyo3(text_signature = "($self, location, start, length)")]
-    fn get_range(&self, location: PyPath, start: usize, length: usize) -> PyResult<Py<PyBytes>> {
+    fn get_range(&self, location: PyPath, start: usize, length: usize) -> PyResult<Cow<[u8]>> {
         let range = std::ops::Range {
             start,
             end: start + length,
         };
         let obj = self
             .rt
             .block_on(self.inner.get_range(&location.into(), range))
-            .map_err(ObjectStoreError::from)?
-            .to_vec();
-        Python::with_gil(|py| Ok(PyBytes::new(py, &obj).into_py(py)))
+            .map_err(ObjectStoreError::from)?;
+        Ok(Cow::Owned(obj.to_vec()))
+    }
+
+    /// Return the bytes that are stored at the specified location in the given byte range
+    #[pyo3(text_signature = "($self, location)")]
```

Comment:
Is this missing the start/length parameters

Reply:
Yeah
```suggestion
    #[pyo3(text_signature = "($self, location, start, length)")]
```